2.3 -> 3.0: Merge pull request #5417 from chrisvest/2.3-lost-update-pc-test

Fix flaky page cache tests

jakewins opened the pull request:
## 2.3 -> 3.0 merge

This is an automated PR created whenever there are commits in the `2.3` branch that are not in the `3.0` branch.

If this PR is green, feel free to merge it. If it is not green, or it has conflicts, help make sure those errors are owned and on their way to be resolved.

If this job is causing issues, ping Jake.

chrisvest committed:
StoreFactory cleanup

 * Reduce method visibility modifies where possible
 * Remove unused Monitors arguments from many constructors and their
   invocations

chrisvest committed:
Split NeoStores and MetaDataStore

- This introduces the NeoStores concept, as the class responsible for holding
  references to the other stores.
- The NeoStores is also in charge of initialising the other stores, and does so
  lazily, so tools can continue to work on incomplete store directories.
- The stores themselves are now in charge of creating their store files, and
  writing headers and trailers to them.
- The StoreFactory has been reduced to something that gathers up configuration
  information, and then creates the NeoStores.
- NeoStore has been renamed to MetaDataStore, though the file itself is still
  called "neostore".
- The responsibility of closing stores has also been migrated from the
  individual stores, into the NeoStores.

rickardoberg committed:
Support for hostnames in clustering
Payload now writes an int instead of byte for message length. This breaks backwards compatibility with 2.2, and so rolling upgrades between 2.2 and 2.3 will not work.

SRaghuram committed:
Faster consistency checker

Basically by reducing random access using an effient cache, checking only
parts of records through multiple passes. Also parallelizes some store
checks, like indexes.

apcj committed:
EXPLAIN canned queries and check for notifications.

akollegger committed:
new drawer for changing configuration

davidegrohmann committed:
Add Metrics Kernel Extension

tinwelint committed:
Removes more remnants of diff check

technige committed:
PackStream map streams

technige committed:
List streams too

tinwelint committed:
Brings in the legacy consistency checker as a separate component

which will live in community/consistency-check-legacy until we're
confident that the new and big changes are stable and works in all
scenarios. The legacy checker is a complete copy of the code base as it
was in 2.3 HEAD right before this commit, and as such can be deleted
without affecting the new checker when that time comes.

The legacy checker will be default for a little while longer, but can be
run if --experimental is supplied.

Up to date with commit be67faa2c0942071705b11828ca0d9753be67d36

Commit:
Started adding MergeInto

srbaker committed:
Set properties on merge.

chrisvest committed:
Make it possible to build from a git-worktree checkout

henriknyman committed:
Improved performance of load csv with headers

Instead of creating a new scala map for each row, we use an array based map where we
only update the values once for every row.

davidegrohmann committed:
Remove trailers from store files since they are redundant

They are _not_ used to determine recovery and they are _not_ useful
for rebuilding id generators since ".id" files have their own way to
check if they were cleanly closed or not.

This commit removes the trailers from the stores.

Note: the counts store relies on the trailer to determine if the
latest version was completely flushed on disk or not.  In order to
preserve this check now we write as trailer the "format specifier"
which can be found in the header as well.

For future reference: Trailers are used before migration to check that
the stores are closed cleanly, since we have trailers in older store
previous 2.3, this is fine.  For post and 2.3 stores the best way of
implementing this would be to check that the latest entry in the
transaction log is a checkpoint.

digitalstain committed:
Revert "Merge branch '2.2' into 2.3"

This reverts commit db860849202aaeba5c3ea9ee6887cb457ee645bc, reversing
changes made to 4c307e37324dd14913924fb21b81998ee7a1e0d4.

Conflicts:
	community/kernel/src/test/java/org/neo4j/kernel/impl/locking/LegacyDeadlockCompatibility.java

digitalstain committed:
Merge pull request #5313 from digitalstain/2.3

Revert "Make deadlocks less cryptic, cut rate of aborted transactions", for 2.3

davidegrohmann committed:
Merge 2.2 into 2.3

henriknyman committed:
Generate warning on LOAD CSV with Eager operator

When a LOAD CSV is followed by an Eager operator that forces all data
to be materialized in memory a warning about potential problems
with memory consumption is generated.

pontusmelke committed:
Allow expressions for URL in LOAD CSV

jakewins committed:
Merge pull request #5288 from nigelsmall/2.3-bolt-map-stream

PackStream map streams

jakewins committed:
Merge pull request #5314 from nawroth/2.3-javadocs

2.3 javadocs

jakewins committed:
Merge branch '2.2' into 2.3

Conflicts:
	enterprise/neo4j-enterprise/pom.xml

nawroth committed:
Javadoc compliance with Java 8 for 2.3

thobe committed:
Merge pull request #5206 from akollegger/forbidden-security-rule

new ForbiddenSecurityRule to deny authenticated but forbidden access

thobe committed:
Add deprecated cache_type setting

This is useful for backwards compatibility with older versions of Neo4j.
If the deprecated cache_type setting is used, Neo4j will log a warning.

davidegrohmann committed:
Enable DEBUG logging by default for some modules

jexp committed:
Added Documentation for the Eager pipe

lassewesth committed:
Updated Powershell documentation

Added note about unblocking downloaded files.

pontusmelke committed:
Warnings for LOAD CSV + large non-indexed label

davidegrohmann committed:
Merge 2.2 into 2.3

davidegrohmann committed:
Merge upstram/2.3 into 2.3

chrisvest committed:
Moved the MemoryManager from the io module to the unsafe module and make it public

Because the page cache might not be the only thing that is interested in allocating alligned memory.

thobe committed:
Merge pull request #5325 from nawroth/2.3-java8

Javadoc compliance with Java 8 for 2.3

henriknyman committed:
Merge pull request #5324 from jexp/eager-pipe-docs

Added Documentation for the Eager pipe

cleishm committed:
Move check of URL for LOAD CSV inside QueryContext

Allows for more sophisticated checking to be implemented

cleishm committed:
Fix package for cypher.internal.spi.v2_3 tests

rickardoberg committed:
Use transaction template and retries to handle master switches during test
Add a consumer way of executing templates, which do not require any results

pontusmelke committed:
Merge pull request #5334 from cleishm/2.3-cypher-spi-package-fix

Fix package for cypher.internal.spi.v2_3 tests

Commit:
Merge pull request #5301 from pontusmelke/2.3-merge-on-create-eagerness

Less eagerness when multiple merge statements

Mats-SX committed:
Update CHANGES.txt for Cypher

pgpv committed:
fixed problem with installing different versions of neo4j-desktop

pgpv committed:
Merge branch '2.3' of github.com:neo4j/neo4j into 2.3

davidegrohmann committed:
Merge pull request #5328 from chrisvest/2.3-move-mm-to-unsafe

Moved the MemoryManager from the io module to the unsafe module and make it public

tinwelint committed:
Merge pull request #5323 from davidegrohmann/2.3-enable-debug-logging

Enable DEBUG logging by default for some modules

MishaDemianenko committed:
Merge pull request #5322 from thobe/2.3-deprecate-cache-type

Add deprecated cache_type setting

pontusmelke committed:
Clean up and added check for merge

- moved checks into separate files
- wrote unit tests for checkers
- added warnings for merge

Commit:
Clean up of MergeIntoPipe

pgpv committed:
fixed missing exe

pgpv committed:
fixed code signing problem

davidegrohmann committed:
Merge 2.2 into 2.3

Commit:
Merge pull request #5312 from henriknyman/2.3-load-csv-improvements

Improved performance of load csv with headers

Commit:
Merge pull request #5316 from henriknyman/2.3-warn-eager-loadcsv

LOAD CSV warnings

lassewesth committed:
cosmetics for install4j

tinwelint committed:
CHANGELOGs include kernel changes

tinwelint committed:
Merge pull request #5298 from apcj/tokens

Restructure token handling to allow for other implementations.

digitalstain committed:
Merge branch '2.2' into 2.3

Conflicts:
	community/io/src/main/java/org/neo4j/io/fs/FileLock.java
	community/kernel/src/test/java/org/neo4j/kernel/impl/util/StringLoggerTest.java
	enterprise/ha/src/main/java/org/neo4j/kernel/ha/transaction/TransactionPropagator.java

digitalstain committed:
Merge branch '2.2' into 2.3

Conflicts:
	enterprise/cluster/src/main/java/org/neo4j/cluster/ProtocolServer.java
	enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberStateMachineTest.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/SwitchToSlaveTest.java

pgpv committed:
fixed double version

pgpv committed:
Merge branch '2.3' of github.com:neo4j/neo4j into 2.3

pontusmelke committed:
Updated documentation and CHANGES.txt

Commit:
Fix windows certficate problem
Temporary fix for not creating certificate files if ssl is not enabled

oskarhane committed:
Remove trailing spaces and unused code.

oskarhane committed:
Move initial command to personal

- Make it of type 'string'
- Fix heartbeat model

oskarhane committed:
Update CHANGES.txt for browser

oskarhane committed:
Merge pull request #5343 from oskarhane/2.3-browser-changes

Update CHANGES.txt for browser

Commit:
Merge pull request #5342 from zhenlineo/windows-cert

Fix windows certficate problem

pontusmelke committed:
Planning support for MergeInto

davidegrohmann committed:
Increase timeout for a flaky page cache test

MishaDemianenko committed:
Use specific method from java.lang.System to get line separator.

MishaDemianenko committed:
Import tool help/usage message should provide usage example

Update import tool help message to be printed out when no input parameters were provided.
Text updated to include usage example.

davidegrohmann committed:
Cleanup test by using the ClusterRule

pe4cey committed:
Rework copy in browser guides

lutovich committed:
Merge branch 2.2 into 2.3

Conflicts:
	community/kernel/src/test/java/org/neo4j/kernel/impl/api/integrationtest/ConstraintsCreationIT.java

lutovich committed:
Added unit test for constraints-changed transient error

digitalstain committed:
Merge branch '2.2' into 2.3

Conflicts:
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/TxPushStrategyConfigIT.java

tinwelint committed:
Merge remote-tracking branch 'upstream/2.2' into 2.3

Conflicts:
	enterprise/ha/src/main/java/org/neo4j/kernel/impl/ha/ClusterManager.java
	enterprise/ha/src/test/java/jmx/HaBeanIT.java
	enterprise/ha/src/test/java/org/neo4j/ha/ClusterTransactionTest.java
	enterprise/ha/src/test/java/org/neo4j/ha/HaCacheIT.java
	enterprise/ha/src/test/java/org/neo4j/ha/TestBranchedData.java
	enterprise/ha/src/test/java/org/neo4j/ha/TestClusterClientPadding.java
	enterprise/ha/src/test/java/org/neo4j/ha/TestPullUpdates.java
	enterprise/ha/src/test/java/org/neo4j/ha/TestSlaveOnlyCluster.java
	enterprise/ha/src/test/java/org/neo4j/kernel/HACountsPropagationTest.java
	enterprise/ha/src/test/java/org/neo4j/kernel/api/SchemaIndexHaIT.java
	enterprise/ha/src/test/java/org/neo4j/kernel/api/UniqueConstraintHaIT.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/ClusterTopologyChangesIT.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/DeletionTest.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/HaLoggingIT.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/SlaveUpdatePullerTest.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/TxPushStrategyConfigIT.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/UniqueConstraintStressIT.java
	enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/member/HighAvailabilitySlavesIT.java
	enterprise/ha/src/test/java/org/neo4j/kernel/index/AutoIndexConfigIT.java
	enterprise/ha/src/test/java/org/neo4j/test/ha/ClusterRule.java
	enterprise/ha/src/test/java/org/neo4j/test/ha/ClusterTest.java

lutovich committed:
Restored test removed by mistake

UniquenessConstraintCreationIT was deleted my mistake during PEC restructuring.
This commit restores it and makes some small cleanup in PEC related tests.

lutovich committed:
Removed PMD maven plugin

cleishm committed:
Allow URL access checking to be customized at construction

davidegrohmann committed:
Merge pull request #5354 from lutovich/2.3-restore-test

Restored test removed by mistake

davidegrohmann committed:
Merge pull request #5350 from MishaDemianenko/2.3-line-separator-cleanup

Use specific method from java.lang.System to get line separator.

oskarhane committed:
Add to CHANGES.txt

oskarhane committed:
Merge pull request #5253 from oskarhane/2.3-update-existing-guides

Update existing guides in browser

oskarhane committed:
Merge pull request #5309 from akollegger/browser-config-panel

new drawer for changing configuration

lutovich committed:
Merge branch 2.2 into 2.3

Commit:
Fix the error that cannot create certs on windows

jakewins committed:
Merge pull request #5355 from lutovich/2.3-remove-pmd

Removed PMD maven plugin

pontusmelke committed:
Support for gzip and zip files for LOAD CSV

chrisvest committed:
Expose Unsafe.copyMemory

MishaDemianenko committed:
Restore TestClusterClientPadding test. Switch to cluster rule usage.

chrisvest committed:
Make sure to always close any opened PageSwapper in the PageSwapper tests

lutovich committed:
Merge pull request #5349 from MishaDemianenko/2.3-import-message

Import tool help/usage message should provide usage example

pgpv committed:
Merge pull request #5358 from zhenlineo/windows-cert-change

Fix the error that cannot create certs on windows

chrisvest committed:
Do not over-specify the return value of vectored read beyond end of file

cleishm committed:
Add `dbms.security.load_csv_file_url_root` setting

Which restricts file URLs used with `LOAD CSV` to be within the
specified directory, such that the URL is resolved relative
to that root directory (similarly to a chroot).

There is currently no default for the setting, which allows file URLs to
access any local file. For neo4j-server, the default configuration files
explicitly set this to `data/import`, so new deployments will be
restricted by default. The `data/import` directory has also been added
to the server packaging.

For Neo4j 3.0, a default should be added to this property, so all
deployments are made safe by default.

jakewins committed:
Merge pull request #5360 from chrisvest/2.3-memcpy

Expose Unsafe.copyMemory

jakewins committed:
Merge pull request #5362 from chrisvest/2.3-close-swappers

Make sure to always close any opened PageSwapper in the PageSwapper tests

chrisvest committed:
Merge pull request #5057 from chrisvest/2.3-more-store-create

StoreFactory cleanup

davidegrohmann committed:
Merge pull request #5361 from MishaDemianenko/2.3-cluster-client-padding

Restore TestClusterClientPadding test. Switch to cluster rule usage.

pontusmelke committed:
LOAD CSV support gzip and deflate over http

pontusmelke committed:
Updated changes.txt

pontusmelke committed:
Added documentation about the new operator

MishaDemianenko committed:
Query log rotation

Add possibility to rotate query. Rotation threshold is specified by 'dbms.querylog.rotation.threshold' property
and maximum number of history files to keep by 'dbms.querylog.max_archives'.
By default rotation will happen each 20m and we will keep 7 files.

Mats-SX committed:
Merge pull request #5317 from pontusmelke/2.3-load-csv-expressions

Allow expressions for URL in LOAD CSV

chrisvest committed:
Add neo4j-metrics to the enterprise distribution

Also add an integration test for the metrics kernel extension.

chrisvest committed:
Git-ignore a couple of neo4j-home directories that are created during build

chrisvest committed:
First attempt at adding a page to the manual about the metrics plugin

pontusmelke committed:
Warn when user provides empty labels

nawroth committed:
Fixing metrics docs.

digitalstain committed:
Merge branch '2.2' into 2.3

pontusmelke committed:
Warn when relationship type is missing

pontusmelke committed:
Warn for missing property key names

digitalstain committed:
Merge pull request #5273 from rickardoberg/2.3-resolve-hostnames

Cluster code no longer resolves hostnames into IPs.

davidegrohmann committed:
Merge pull request #5365 from cleishm/2.3-url-access-rules

Security improvements to file URL access in `LOAD CSV`

chrisvest committed:
Polishing the metrics documentation

tinwelint committed:
Fewer calls to NeoStores#getXXXStore()

Instead getting the required store(s) ones and use directly. This is a
good practive, for avoiding that additional method call getting
the store for every interaction.

Before this commit there was additional synchronization on each getter
since stores were openened lazily on first access. This was particularly
noticable in hot execution paths. This is still supported, but the default
behaviour is to eagerly open all stores and if choosing that option
removing the synchronization overhead.

There are still some places where it's very convenient to not pull out
each store, since all stores are used, f.ex. when applying transactions
onto the store. Those are left intact.

tinwelint committed:
Merge remote-tracking branch 'upstream/2.2' into 2.3

Null merge

tinwelint committed:
Merge pull request #5372 from MishaDemianenko/2.3-query-log-rotation

Query log rotation

davidegrohmann committed:
Merge 2.2 into 2.3

chrisvest committed:
Clean up the metrics code a little bit

Mats-SX committed:
Merge pull request #5363 from pontusmelke/2.3-load-csv-compress

Support for gzip, zip and deflate headers for LOAD CSV

chrisvest committed:
More metrics clean up

henriknyman committed:
Allow expressions in SKIP/LIMIT

- Changed parser so expressions are allowed in SKIP/LIMIT
- Added semantic check that fails if the expression depends on identifiers
- Added semantic check that fails if literals are not positive integers

chrisvest committed:
Introduce a Counter abstraction and use it to make ByteCountsMetric thread-safe

davidegrohmann committed:
Tentative fix for the Window build

chrisvest committed:
Clean up from constants in metrics

chrisvest committed:
Restore pom.xml groupId tags that Intellij thinks are redundant

chrisvest committed:
Mention the new metrics component in CHANGES.txt

pgpv committed:
fixed wrong icon alignment for mac installer

lutovich committed:
Merge pull request #5280 from tinwelint/2.3-faster-cc

Faster consistency checker

pe4cey committed:
Update cypher examples to not return warnings

cleishm committed:
Remove inadvertently commited test

Commit:
Merge pull request #5346 from pontusmelke/merge-into

New operator MergeInto

pontusmelke committed:
Updated documentation

Commit:
Merge pull request #5378 from pontusmelke/2.3-warn-for-empty

Warnings for unavailable labels, relationship types, and properties

chrisvest committed:
Increase the semi-long timeout for the page cache tests to account for variable IO latency

oskarhane committed:
Merge pull request #5386 from pe4cey/2.3-guides-produce-warnings

2.3 guides produce warnings

pontusmelke committed:
Fix for failing windows build

davidegrohmann committed:
Fix loads csv tests on windows

chrisvest committed:
Straighten out the life cycling of the metrics extension

This makes the StoreMigrator* tests pass, because the metrics extension will now skip loading by throwing an exception (about missing dependencies) at the correct time in its life cycle

davidegrohmann committed:
Merge pull request #5392 from pontusmelke/2.3-failing-windows

Fix for failing windows build

pe4cey committed:
Ignore warnings in notifications if type is PropertyNameMissingWarning

davidegrohmann committed:
Number of processors should never be zero in Consitency checker

This was causing a division by zero when there is only one processor

chrisvest committed:
Add description annotations on the MetricsSettings

davidegrohmann committed:
CheckPointer return the transaction id used in during check pointing

Mats-SX committed:
Update CHANGES.txt for Cypher

Mats-SX committed:
Merge pull request #5381 from henriknyman/2.3-skip-operator-with-expressions

Allow expressions in SKIP/LIMIT

pe4cey committed:
Taking canned cypher tests offline

pe4cey committed:
Using the correct Ignore import

Intelli-fail auto-import on this one.

tinwelint committed:
Work-around for jUnit run issue in Eclipse

chrisvest committed:
Remove org.neo4j.concurrent.Counters

chrisvest committed:
Update com.codahale.metrics dependency to latest 3.1.2

chrisvest committed:
More metrics cleanup

Including removing a metric (logdeserializer) that had not been working at least since the JTA removal, but probably didn't work before that either.

Mats-SX committed:
Remove an implementation from QueryContext

Mats-SX committed:
Force which node the rule planner will expand a merge pattern from

Previously, a merge pattern would always be expanded from the node which had
an identifier that was returned first by a Map.values call. This order is not
well defined. From now on identifiers are sorted in ascending lexicographical
order and the first bound one is picked.

chrisvest committed:
Merge pull request #5305 from davidegrohmann/2.3-remove-store-trailers

Remove trailers from store files since they are redundant

davidegrohmann committed:
Add tryCheckPoint to CheckPointer

TryCheckPoint should check point if no check pointing is currently
running, otherwise it will wait for the running check point to
complete and it will not exexcute any check pointing.

All the check point methods are now synchronized to ensure that there is always
only one thread performing a check point.

davidegrohmann committed:
StoreCopyServer checkpoint before full backup

Instead of flushing store files on disk and cause a race with the
check pointer, it will ask the check pointer to try check point for
making sure we have flushed store files on disk, and after that it
will start streaming transactions starting from the transaction used
in the check point.

This should solve a failure caused by the race mentioned above where
we were trying to rotate the counts store while it was in a rotation
state.

pe4cey committed:
Copy changes in :help and info drawer

chrisvest committed:
Merge branch '2.2' into 2.3

thobe committed:
Merge 2.2 into 2.3

thobe committed:
Remove duplicate module from pom

tinwelint committed:
Merge remote-tracking branch 'upstream/2.2' into 2.3

Conflicts:
	community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java
	community/kernel/src/main/java/org/neo4j/kernel/impl/api/TransactionRepresentationCommitProcess.java
	community/kernel/src/main/java/org/neo4j/kernel/impl/api/TransactionRepresentationStoreApplier.java
	community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexUpdatesValidator.java
	community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java
	community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/Commitment.java
	community/kernel/src/test/java/org/neo4j/kernel/impl/api/index/IndexRecoveryIT.java
	community/kernel/src/test/java/org/neo4j/kernel/impl/api/index/OnlineIndexUpdatesValidatorTest.java
	community/kernel/src/test/java/org/neo4j/kernel/impl/transaction/state/NeoStoreTransactionTest.java
	enterprise/backup/src/main/java/org/neo4j/backup/BackupService.java
	enterprise/backup/src/main/java/org/neo4j/backup/RebuildFromLogs.java
	enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java
	enterprise/com/src/test/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpackerTest.java
	enterprise/ha/src/main/java/org/neo4j/kernel/ha/HaSettings.java
	enterprise/ha/src/main/java/org/neo4j/kernel/ha/HighlyAvailableGraphDatabase.java
	enterprise/ha/src/test/java/org/neo4j/ha/upgrade/MasterClientTest.java
	enterprise/query-logging/src/main/java/org/neo4j/kernel/impl/query/QueryLoggerKernelExtension.java
	enterprise/query-logging/src/test/java/org/neo4j/kernel/impl/query/QueryLoggerIT.java

tinwelint committed:
Merge remote-tracking branch 'upstream/2.3' into 2.3

tinwelint committed:
Merge pull request #5380 from tinwelint/2.3-fewer-get-store-calls

Fewer calls to NeoStores#getXXXStore()

davidegrohmann committed:
Add prints to help debug flaky test in CI

To be removed as soon as the test is fixed

tinwelint committed:
Passes in correct dependencies to TransactionCommittingResponseUnpacker

cleishm committed:
Use absolute URLs and load_csv_file_url_root in feature suite

Rather than using non-standard absolute URLs, we can now use
absolute URLs and the load_csv_file_url_root setting to
control where they are sourced from.

This resolves a windows issue mapping those non-standard URLs
to java Paths

nawroth committed:
Fix build problems related to neo4j-harness-test

jexp committed:
Allow custom Accept-Headers with HTTP utility

MishaDemianenko committed:
Incremental backup failure handling and logging update

Add additional logging statement to see why incremental backup fail.
Update page cache error handling.

davidegrohmann committed:
Merge 2.2 into 2.3

pontusmelke committed:
Don't use List when it doesn't make sense

chrisvest committed:
Add generated documentation for the MetricsSettings to the manual

chrisvest committed:
Fix test isolation failure in PageCacheTest.mustNotLoseUpdates

The test uses the RepeatRule to run multiple times, and was using a file with a fixed name.
It was relying on the ephemeral file system being recreated every run, which is not neccesarily the case for the tests that uses the real file system.
This is fixed by explicitly zeroing out the file pages before we use them.

pontusmelke committed:
Merge pull request #5415 from pontusmelke/2.3-url-fix

Use absolute URLs and load_csv_file_url_root in feature suite

cleishm committed:
Merge branch 2.2 into 2.3

tinwelint committed:
Merge pull request #5403 from davidegrohmann/2.3-checkpoint-when-full-backup

Check point before doing a full backup

nawroth committed:
Fixes for Metrics docs and settings docs tooling

Mats-SX committed:
Merge pull request #5407 from pontusmelke/2.3-replace-linear-list-usages

Do not use toList where toVector is a better option

lutovich committed:
Merge pull request #5417 from chrisvest/2.3-lost-update-pc-test

Fix flaky page cache tests

Commit:
Updated packaged config files for the new cert-file properties

When Neo with default starts up it complains about outdated cert-location properties. This should not happen with the built in config

chrisvest committed:
Use buffered IO when writing CSV metrics files

MishaDemianenko committed:
Merge pull request #5419 from nawroth/2.3-metrics-plugin

Metrics plugin

MishaDemianenko committed:
Fix newly introduced checkpointer tests

Fix races between different checkpointering threads.

pgpv committed:
fixed ce in artifact names and in neo desktop itself

pgpv committed:
fixed problem with spaces in the path for the neo desktop mac command prompt

pgpv committed:
Merge branch '2.3' of github.com:neo4j/neo4j into 2.3

Commit:
Merge remote-tracking branch 'origin/2.2' into 2.3

pgpv committed:
Merge branch '2.3' of github.com:neo4j/neo4j into 2.3

tinwelint committed:
Merge pull request #5424 from MishaDemianenko/2.3-java8-failed-test

Fix newly introduced checkpointer tests

pontusmelke committed:
Fix in manual for Pretty Graphs

tinwelint committed:
Always call makeStoreOk from NeoStores since stores no longer delegate internally

tinwelint committed:
Merge pull request #5430 from tinwelint/2.3-centralize-makestoreok

Always call makeStoreOk from NeoStores since stores no longer delegate internally

akollegger committed:
intercom messaging can be used even when udc reporting not enabled

davidegrohmann committed:
Merge pull request #5412 from jexp/custom-http-accept-headers

Allow custom Accept-Headers with HTTP utility

craigtaverner committed:
Merge pull request #5431 from pontusmelke/2.3-fix-pretty-graphs

Fix in manual for Pretty Graphs

pgpv committed:
another naming change

pgpv committed:
code signing

pgpv committed:
Merge branch '2.3' of github.com:neo4j/neo4j into 2.3

Commit:
Introduce STARTS WITH, ENDS WITH and CONTAINS

lutovich committed:
Merge branch 2.2 into 2.3

Conflicts:
	community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/Status.java
	community/server/CHANGES.txt
	community/server/src/test/java/org/neo4j/server/rest/AbstractRestFunctionalTestBase.java

oskarhane committed:
Merge pull request #5434 from akollegger/browser-intercom-message-button

intercom messaging can be used even when udc reporting not enabled

lutovich committed:
Fixed broken SessionStateMachineTest

Failure was caused by the commit that changed ClientError's effect on
transaction to be ROLLBACK instead of NONE.

nawroth committed:
Put libraries in the right location of packages

* Put all Scala libraries in lib/.
* Put the dependencies of neo4j-metrics in lib/
  slf4j-api is one such transitive dependecy, so it got moved.
* Clean up of assembly descriptors to avoid warnings.
* Don't use a property for metrics.version.
  It caused strange issues with Maven.

chrisvest committed:
Don't pre-zero freshly allocated memory.

This is safe because it will always be overwritten or zeroed out when we page fault.

chrisvest committed:
Add a hidden configuration for the MemoryManager grab size

davidegrohmann committed:
Merge pull request #5440 from nawroth/2.3-metrics-packaging

Put libraries in the right location of packages

Mats-SX committed:
Add tests for whitespace arguments

Mats-SX committed:
Change prefix comparison to 1/(length+1)

Added a constant for default value (unknown prefix length).

Mats-SX committed:
Remove traces of LIKE

Add to combined CHANGES.txt files.

Commit:
Merge branch 2.2 into 2.3

glennsarti committed:
GitHub Issue #5326

The Get-Neo4jServer function was using the literal input from the user without resolve to its true absolute path.  This created issues with items like trailing slashes in the path.  The changes resolve the path to the Neo4j installation to an absolute path name without trailing slashes.  Tests were added to test various scenarios

glennsarti committed:
Created a Powershell Module to manage a Neo4j Server

Modified Confirm-Neo4jServerObject to compare only string values.  This means it will detect empty strings and null values as invalid
Modified the ShouldProcess in Start-Neo4jArbiter and Start-Neo4jServer with the correct parameters used to execute the java process
Added examples to Set-Neo4jSetting and Remove-Neo4jSetting

Mats-SX committed:
Removed superfluous warning for index lookups

Previously we differentiated warnings for not being able to use index seeks and
index scans when using dynamic property lookup. This commit merges the
warnings into one.
Reduced a parameter to speed up long-running test.

Commit:
Make sure the prettifier handles STARTS WITH and ENDS WITH

chrisvest committed:
Fix page cache memory reporting when `dump_configuration` is enabled

The previous code did integer multiplication of the number of pages in the cache, and the page size, which would overflow if the page cache was given more than 2 GiBs of memory.
This does not change how the page cache memory allocation is calculated, which was already correct, it only changes how this allocation is reported by `dump_configuration`.

Commit:
Cleanup

Commit:
Merge pull request #5401 from Mats-SX/2.3-document-merge-order

Force which node the rule planner will expand a merge pattern from

pontusmelke committed:
Merge pull request #5447 from Mats-SX/2.3-one-warning-dyn-prop-lookup

Removed superfluous warning for index lookups

lutovich committed:
Fixed flaky TransientErrorIT

tinwelint committed:
Merge branch '2.2' into 2.3

Conflicts:
	community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java
	community/kernel/src/main/java/org/neo4j/kernel/impl/store/DumpStore.java

tinwelint committed:
Merge remote-tracking branch 'upstream/2.3' into 2.3

thobe committed:
Merge pull request #5442 from chrisvest/2.3-pagecache-no-pre-zeroing

Don't pre-zero freshly allocated memory.

zhenlineo commented:
I would close this one once this merge https://github.com/neo4j/neo4j/pull/5452 got in.

tinwelint committed:
Merge remote-tracking branch 'upstream/2.2' into 2.3

Conflicts:
	community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreFile.java
	enterprise/backup/src/main/java/org/neo4j/backup/BackupService.java
	enterprise/backup/src/test/java/org/neo4j/backup/TestBackup.java

lutovich committed:
Removed printlns from ShortestPathAcceptanceTest

pontusmelke committed:
Remove hidden upper limit in shortestPath

`shortestPath` when provided with a unbounded pattern will use an upper limit
of 15 hops. This PR removes this limit and instead provides a warning when using
`shortestPath` with an unbounded pattern.

pontusmelke committed:
Merge pull request #5441 from systay/startswit

Introduce STARTS WITH, ENDS WITH and CONTAINS

tinwelint committed:
Merge pull request #5449 from lutovich/2.3-flaky-TransientErrorIT

Fixed flaky TransientErrorIT

jotomo committed:
Make prettifier handle DETACH DELETE.

tinwelint committed:
Merge pull request #5436 from MishaDemianenko/2.3-backup-too-late

Incremental backup failure handling and logging update

lutovich committed:
Merge branch 2.2 into 2.3

Commit:
Merge pull request #5461 from jotomo/2.3-prettify-detach-delete-in-manual

Make prettifier handle DETACH DELETE.

tinwelint committed:
Removed duplicate which now lives in src/main

nawroth committed:
Fix bug that removes the live console

tinwelint committed:
Merge pull request #5421 from zhenlineo/update-cert-in-config-file

Updated packaged config files for the new cert-file properties

tinwelint committed:
Merge remote-tracking branch 'upstream/2.2' into 2.3

tinwelint committed:
Merge branch '2.2' into 2.3

Conflicts:
	enterprise/ha/CHANGES.txt

digitalstain committed:
Merge branch '2.2' into 2.3

Conflicts:
	community/kernel/src/main/java/org/neo4j/kernel/InternalAbstractGraphDatabase.java
	community/kernel/src/test/java/org/neo4j/kernel/InternalAbstractGraphDatabaseTest.java

nawroth committed:
Merge remote-tracking branch '2.2' into 2.3

lassewesth committed:
Merge pull request #5446 from glennsarti/feature/powershell

Resolved github issue 5326

Mats-SX committed:
Merge pull request #5454 from pontusmelke/2.3-remove-shortest-path-limit

Remove hidden upper limit in shortestPath

pontusmelke committed:
Merge remote-tracking branch 'upstream/2.2' into 2.3

oskarhane committed:
Merge pull request #5406 from pe4cey/update-information-sidebar

Copy changes in :help and info drawer

zhenlineo commented:
a marker to specify where I am in this merge train https://github.com/neo4j/neo4j/pull/5468

nawroth committed:
Merge remote-tracking branch 'nawroth/2.3-fix-refcard' into 2.3

nawroth committed:
Merge remote-tracking branch '2.2' into 2.3

nawroth committed:
Merge pull request #5463 from nawroth/2.3-fix-live-console

Fix bug that removes the live console

pontusmelke committed:
Fix failing build
